actors/verifreg/src/types.rs

```diff
@@ -168,18 +169,33 @@ pub struct AllocationRequest {
 }
 impl Cbor for AllocationRequest {}
 
+// A request to extend the term of an existing claim with datacap tokens.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct ClaimExtensionRequest {
+    pub provider: Address,
+    pub claim: ClaimID,
+    pub term_max: ChainEpoch,
+}
+impl Cbor for ClaimExtensionRequest {}
+
 /// Operator-data payload for a datacap token transfer receiver hook specifying an allocation.
 /// The implied client is the sender of the datacap.
 #[derive(Clone, Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
 pub struct AllocationRequests {
-    pub requests: Vec<AllocationRequest>,
+    pub allocations: Vec<AllocationRequest>,
```

Comment:
Purely musing out loud, at what point do we overlay virtual method calling on top of the universal receiver hook?  Maybe the next additional functionality pushes it over the edge.  I expect there is precedent in other smart contract platforms.  It's late to get this in for nv17 but this is a good place to demonstrate how to do it if it's a pattern we want to showcase in the same way we are showcasing token interface with datacap actor.

Reply:
An immediate design question: do you do dispatch on internal functions that stay within the method or do you delegate to other methods on the actor and use the actual vm to do the work for you?